File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit=0;
        int maxi=prices[0];
        for(int i=1;i<prices.length;i++){
            if(prices[i]-maxi>profit)profit=prices[i]-maxi;
            else if(prices[i]<maxi)maxi=prices[i];
        }
        return profit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit=0,buy=prices[0];
        for(int i=1;i<prices.size();i++){
            profit=max(profit,prices[i]-buy);
            buy=min(buy,prices[i]);
        }
        return profit;
    }
};